### PR TITLE
chore: use es6 arrows function

### PR DESCRIPTION
**Pull request by SimonSchick**
This converts most function callback use to arrow functions.

I will push 3 invidual commits and run them through CI (builds are extremely flaky locally).
The first 2 will run CI without lint, the 3rd one adds linting back.

The first commit was entirely automated via eslint.
The second and 2nd is fixing docs and invalid-this errors.
The 3rd final invalid-this errors.

### CODE REVIEW

**Comment by sidorares** [comment]
> On another note: I changed the way the db awaiter works, less spam :)

Thanks! One more thing to improve with test signal/noise ratio is to clean up "packet out of order" messages from test server (or actually fix packet numbers in server)

**Comment by SimonSchick** [comment]
Yea, that's out of scope of my PR tho, plus I'm not exactly sure how I would fix those 😛

**Comment by SimonSchick** [comment]
rebased

**Comment by SimonSchick** [comment]
Broke link, goddamnit 😞

**Comment by SimonSchick** [comment]
Hey @sidorares would you like to merge this and resolve conflicts yourself, or should I rebase again?

**Comment by sidorares** [comment]
if you can would be great, otherwise I'll do

**Comment by SimonSchick** [comment]
Done, hoping that tests pass :)

File: lib/pool_connection.js [hunk 1]
```diff
@@ -42,9 +43,9 @@ class PoolConnection extends Connection {
     this.release();
   }
 
-  destroy() {
+  destroy(...args) {
```
Comment (sidorares) [inline lib/pool_connection.js]: this fails in node 4

Comment (SimonSchick) [reply lib/pool_connection.js]: Aww snap, sorry I've been testing on v8 locally 😛

Comment (sidorares) [reply lib/pool_connection.js]: I'm on v10 locally :)

**Comment by SimonSchick** [comment]
Oh great, I amended without pulling first 😅

**Pull request merged**